packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +869,777 @@ class _CupertinoAlertActionSection extends StatelessWidget {
-            child: new UnconstrainedBox(
-              constrainedAxis: Axis.horizontal,
-              child: new ConstrainedBox(
-                constraints: const BoxConstraints(minHeight: _kButtonHeight),
-                child: new Row(
-                  children: children.map<Widget>((Widget button) {
-                    return new Expanded(child: button);
-                  }).toList(),
-                ),
-              ),
+      ),
+    );
+  }
+}
+
+// Button that updates its render state when pressed.
+//
+// The pressed state is forwarded to an _ActionButtonParentDataWidget. The
+// corresponding _ActionButtonParentData is then interpreted and rendered
+// appropriately by _RenderCupertinoDialogActions.
+class _PressableActionButton extends StatefulWidget {
+  const _PressableActionButton({
+    @required this.child,
+  });
+
+  final Widget child;
+
+  @override
+  _PressableActionButtonState createState() => new _PressableActionButtonState();
+}
+
+class _PressableActionButtonState extends State<_PressableActionButton> {
+  bool _isPressed = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return new _ActionButtonParentDataWidget(
+      isPressed: _isPressed,
+      // TODO:(mattcarroll): Button press dynamics need overhaul for iOS: https://github.com/flutter/flutter/issues/19786
+      child: new GestureDetector(
+        behavior: HitTestBehavior.opaque,
+        onTapDown: (TapDownDetails details) => setState(() => _isPressed = true),
+        onTapUp: (TapUpDetails details) => setState(() => _isPressed = false),
+        // TODO(mattcarroll): Cancel is currently triggered when user moves past slop instead of off button: https://github.com/flutter/flutter/issues/19783
+        onTapCancel: () => setState(() => _isPressed = false),
+        child: widget.child,
+      ),
+    );
+  }
+}
+
+// ParentDataWidget that updates _ActionButtonParentData for an action button.
+//
+// Each action button requires knowledge of whether or not it is pressed so that
+// the dialog can correctly render the button. The pressed state is held within
+// _ActionButtonParentData. _ActionButtonParentDataWidget is responsible for
+// updating the pressed state of an _ActionButtonParentData based on the
+// incoming [isPressed] property.
+class _ActionButtonParentDataWidget extends ParentDataWidget<_CupertinoDialogActionsRenderWidget> {
+  const _ActionButtonParentDataWidget({
+    Key key,
+    this.isPressed,
+    @required Widget child,
+  }) : super(key: key, child: child);
+
+  final bool isPressed;
+
+  @override
+  void applyParentData(RenderObject renderObject) {
+    assert(renderObject.parentData is _ActionButtonParentData);
+    final _ActionButtonParentData parentData = renderObject.parentData;
+    if (parentData.isPressed != isPressed) {
+      parentData.isPressed = isPressed;
+
+      // Force a repaint.
+      final AbstractNode targetParent = renderObject.parent;
```

Comment:
Is this needed instead of just calling markNeedsPaint? Since this is private and there's no repaint boundary, when this paints, the parent will also paint right?

Reply:
"this" is a parent data widget, it doesn't have a render object and so has nothing to paint.

Reply:
Ah I see. Makes sense. My bad.